### ISSUE TITLE
Bug: Exception on analyzing ddtrace package

### ISSUE THREAD

**Issue by MasterOdin** [FHPythonUtils/LicenseCheck#68]
## Before You Begin

Before proceeding, please make sure to follow these steps:

- [x] I have checked for similar issues in the project's issue tracker.
- [x] I have searched closed issues to see if a similar problem was reported
  before.

## Issue Details

When running `licensecheck` on our private repo, it crashes out complaining about not being able to find package metadata for one of the installed dependencies.

### Description

Our `pyproject.toml` dependencies:

```
[tool.poetry.dependencies]
python = "^3.10"
confluent-kafka = "^2.2.0"
dataclasses-json = { version = "^0.5.14", python = ">=3.10,<3.13" }
sqlfluff = "2.3.5"
loguru = "^0.7.0"
memoization = "^0.4.0"
ddtrace = "^1.18.5"
watchdog = {extras = ["watchmedo"], version = "^3.0.0"}
environs = "^9.5.0"

[tool.poetry.group.dev.dependencies]
mypy = "^1.5.1"
pytest = "^7.4.2"
pylint = "^3.0.3"
black = "^23.12.0"
```

### Expected Behavior

That we get all licenses of all dependencies

### Actual Behavior

Seeing the following error:

```
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.7/Frameworks/Python.framework/Versions/3.11/lib/python3.11/importlib/metadata/__init__.py", line 563, in from_name
    return next(cls.discover(name=name))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/licensecheck/packageinfo.py", line 25, in getPackageInfoLocal
    pkgMetadata = metadata.metadata(requirement)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.7/Frameworks/Python.framework/Versions/3.11/lib/python3.11/importlib/metadata/__init__.py", line 998, in metadata
    return Distribution.from_name(distribution_name).metadata
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.7/Frameworks/Python.framework/Versions/3.11/lib/python3.11/importlib/metadata/__init__.py", line 565, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for ENVIER

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/licensecheck/packageinfo.py", line 117, in getPackages
    packageinfo.add(getPackageInfoLocal(requirement))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/licensecheck/packageinfo.py", line 49, in getPackageInfoLocal
    raise ModuleNotFoundError from error
ModuleNotFoundError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/bin/licensecheck", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/licensecheck/__init__.py", line 96, in cli
    myLice, depsWithLicenses = get_deps.getDepsWithLicenses(
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/licensecheck/get_deps.py", line 208, in getDepsWithLicenses
    packages = packageinfo.getPackages(reqs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/licensecheck/packageinfo.py", line 120, in getPackages
    packageinfo.add(getPackageInfoPypi(requirement))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/licensecheck/packageinfo.py", line 77, in getPackageInfoPypi
    license=ucstr(
            ^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/licensecheck/types.py", line 21, in __new__
    return super().__new__(cls, v.upper())
                                ^^^^^^^
AttributeError: 'NoneType' object has no attribute 'upper'
```

where `envier` is coming from [`ddtrace`](https://pypi.org/project/ddtrace/) which our project directly relies on. On removing `ddtrace`, then `licensecheck` runs fine.

## System Information

Please provide the following additional information about your system or
environment:

- Operating System (OS): macOS
- OS Version: 14.2.1
- Project Version (if applicable): 2024

**Comment by FredHappyface** [comment]
Unable to reproduce locally from main branch :(, aim to release in the next week or so

**Comment by FredHappyface** [comment]
Unable to reproduce - please let me know if you're still getting this issue

**Comment by MasterOdin** [comment]
I'm no longer seeing the error on the newest version of licensecheck, though I also cannot reproduce the original issue anymore either so I'm not sure what's going on.

Thanks for the work you've done on this project and following up on this invalid bug report. 😅
